### ISSUE TITLE
Build a data science example

### ISSUE THREAD

**Issue by keithmanville** [usnistgov/dioptra#664]
Demonstrate that Dioptra is a suitable platform for designing and running data science experiments by building a new example around a simple data science use case.

The example should consist of a plugin, entrypoint, and notebook that uses the plugin and entrypoint to walk through through the example with description and visualizations. The example should include a dioptra toml for importing the required resources.

The example will adapt the following:
https://github.com/colton-lapp/DioptraDocumentation/blob/main/DioptraUIExample.md

### Blocked by:
- [ ] #659 

### Definition of Done
- [ ] The notebook walks the user through a complete example
- [ ] The example includes a visual guide that walks a user through the steps in the UI
- [ ] The example is merged into dev

**Comment by colton-lapp** [comment]
Updated:
Currently have basic implementation at [this Jupyter notebook file](https://github.com/usnistgov/dioptra/blob/colton-lapp-data-science-demo/examples/data-science-demo/data_science_demo_plugins.ipynb) 

which follows ideas detailed in this doc [https://docs.google.com/document/d/1mI5q4TUkcQUu8sBZf3LAbTVv8vojKxp7f-BXfJAHHnY/edit?tab=t.0#heading=h.rlizz4eweydy](url)

Awaiting artifact functionality before continuing with progress

**Comment by keithmanville** [comment]
closed by #984
